public/app/core/logs_model.ts

```diff
@@ -683,7 +692,10 @@ export function queryLogsVolume<T extends DataQuery>(
       data: [],
     });
 
-    const subscription = (datasource.query(logsVolumeRequest) as Observable<DataQueryResponse>).subscribe({
+    const queryResponse = datasource.query(logsVolumeRequest);
+    const queryObservable = isObservable(queryResponse) ? queryResponse : from(queryResponse);
+
+    const subscription = queryObservable.subscribe({
```

Comment:
Little bit unsure/confused about this.

`datasource.query` is typed to return either a Promise or an Observable. Previously, if it returned a Promise it would have crashed.

Reply:
i think this change is fine. honestly i also did not know `.query()` can return a `Promise`.

Reply:
i think this change is fine. honestly i also did not know `.query()` can return a `Promise`.